backend/src/main/java/ai/verta/modeldb/batchProcess/CollaboratorResourceMigration.java

```diff
@@ -0,0 +1,236 @@
+
+  public void execute(int recordUpdateLimit) {
+    this.paginationSize = recordUpdateLimit;
+    App app = App.getInstance();
+    if (app.getAuthServerHost() != null && app.getAuthServerPort() != null) {
+      app.setAuthServerHost(app.getAuthServerHost());
+      app.setAuthServerPort(app.getAuthServerPort());
+
+      authService = new AuthServiceUtils();
+      roleService = new RoleServiceUtils(authService);
+    } else {
+      LOGGER.debug("AuthService Host & Port not found, OSS setup found");
+      return;
+    }
+
+    LOGGER.info("Migration start");
+    ModelDBUtils.registeredBackgroundUtilsCount();
+    try {
+      migrateProjects();
+      LOGGER.info("Projects done migration");
+      // migrateRepositories();
+      // LOGGER.info("Repositories done migration");
+    } catch (Exception ex) {
+      if (ex instanceof StatusRuntimeException) {
+        StatusRuntimeException exception = (StatusRuntimeException) ex;
+        if (exception.getStatus().getCode().value() == Code.PERMISSION_DENIED_VALUE) {
+          LOGGER.warn("CollaboratorResourceMigration Exception: {}", ex.getMessage());
+        } else {
+          LOGGER.warn("CollaboratorResourceMigration Exception: ", ex);
+        }
+      } else {
+        LOGGER.warn("CollaboratorResourceMigration Exception: ", ex);
+      }
+    } finally {
+      ModelDBUtils.unregisteredBackgroundUtilsCount();
+    }
+
+    LOGGER.info("Migration End");
+  }
+
+  private void migrateProjects() {
+    LOGGER.debug("Projects migration started");
+    Long count = getEntityCount(ProjectEntity.class);
+
+    int lowerBound = 0;
+    final int pagesize = this.paginationSize;
+    LOGGER.debug("Total Projects {}", count);
+
+    while (lowerBound < count) {
+
+      try (Session session = ModelDBHibernateUtil.getSessionFactory().openSession()) {
+        CriteriaBuilder criteriaBuilder = session.getCriteriaBuilder();
+
+        CriteriaQuery<ProjectEntity> criteriaQuery =
+            criteriaBuilder.createQuery(ProjectEntity.class);
+        Root<ProjectEntity> root = criteriaQuery.from(ProjectEntity.class);
+
+        CriteriaQuery<ProjectEntity> selectQuery =
+            criteriaQuery.select(root).orderBy(criteriaBuilder.asc(root.get("id")));
+
+        TypedQuery<ProjectEntity> typedQuery = session.createQuery(selectQuery);
+
+        typedQuery.setFirstResult(lowerBound);
+        typedQuery.setMaxResults(pagesize);
+        List<ProjectEntity> projectEntities = typedQuery.getResultList();
+
+        if (projectEntities.size() > 0) {
+          Set<String> userIds = new HashSet<>();
+          for (ProjectEntity projectEntity : projectEntities) {
+            userIds.add(projectEntity.getOwner());
+          }
+          LOGGER.debug("Project userId list : " + userIds);
+
+          // Fetch the experiment owners userInfo
+          Map<String, UserInfo> userInfoMap =
+              authService.getUserInfoFromAuthServer(userIds, null, null);
```

Comment:
All of this can be removed.

Reply:
No, it is used to fetch the workspace name so we need userInfo of the project owner. so this is required

Reply:
Could you expand on why we need to load user information? I don't get it

Reply:
We have a common method for fetch the workpaceName from workspaceId `roleService.getWorkspaceDTOByWorkspaceId(` so for that i need a userInfo as an argument to fetch workspaceName.
```
WorkspaceDTO workspaceDTO =
                  roleService.getWorkspaceDTOByWorkspaceId(
                      userInfoMap.get(project.getOwner()),
                      project.getWorkspace(),
                      project.getWorkspace_type());
```

Reply:
https://github.com/VertaAI/modeldb/blob/fe66220573e9ebb12c2781b45257a825db853f0a/backend/src/main/java/ai/verta/modeldb/batchProcess/CollaboratorResourceMigration.java#L124

Reply:
I think getWorkspaceDTOByWorkspaceId looks like a code for getting old workspaces information. Here is what I tried to write for new (in registry): https://github.com/VertaAI/registry/pull/102/files#diff-eb351bf96b15680d857baab17a91a6f29aecacc59d81271080602ed248773a16R1226

Reply:
Updated like your suggestion

Reply:
Revert this. we are used vertaId as a workspace on the project so your logic is not working on MDB.